### ISSUE TITLE
`executable` tests require network connection or cached bundle files

### ISSUE THREAD

**Issue by pkgw** [tectonic-typesetting/tectonic#148]
The various tests in the `executable` module compile basic TeX files and so require that the test environment have either a network connection, or cached bundle files in the test user's home directory. It is possible that neither of these will be true in a pristine, non-networked test box, in which case these tests won't work.

Just skip them if the current user doesn't have a Tectonic cache directory? This gets into broader questions of how to test the range of Tectonic's functionality when doing so requires large support files.

**Comment by Mrmaxmeier** [comment]
I think something in the spirit of the hacky TestBundle from #143 would be useful. While this wouldn't help the download-and-save style testcase it'd be enough to test larger LaTeX packages like tikz or beamer.

I'd really like to get the test coverage up to a level where I'm somewhat comfortable refactoring and making changes in the depths of the engine.
@pkgw how would you approach  these testcases?

**Comment by pkgw** [comment]
I guess my main principles are:

- I want the checkout of the main tectonic repository not to require downloading huge amounts of data
- I want it to be possible to run a very comprehensive test suite without a network connection or any assumptions about the state of the user's home directory

Given those principles, the rough picture I have in mind is that tests of more complex functionality should be optional, only running if local versions of necessary support files are present. These can be hosted online and be easy to download, but not included in the repository by default. (We could even provide a shell script to download the files as needed.)

In most cases, I'd think that the "necessary support files" would be a bundle file just like the one hosted on BinTray right now. In fact, it would probably be easiest to get going by using a local copy of precisely that bundle file. Other support files might be binary inputs and outputs that can be large: PDF, XDV, JPG, OTF files, that sort of thing.

One current problem with this approach is that Tectonic currently has problems when you point it to a local bundle file, because that functionality was implemented before I realized that we really need to generate and cache format files on-the-fly. The way I hacked the format file caching together, it only works when using a web-based bundle with a local disk cache. That limitation is silly and should be fixed. (Was this limitation why you took the approach you did with your TestBundle approach, with the way it emulated the structure of the per-user disk cache?)
